client/question/services/FeedbackGeneratorService.js

```diff
@@ -417,21 +485,82 @@ tie.factory('FeedbackGeneratorService', [
        * @returns {Feedback}
        */
       getFeedback: function(tasks, codeEvalResult, rawCodeLineIndexes) {
+        // Reset all counters but consecutiveSameRuntimeErrorCounter.
+        _resetCounters(EXCLUDE_CONSECUTIVE_SAME_RUNTIME_ERROR);
+        // If the user receives the same error message increment the same error
+        // counter.
+        if (previousErrorString &&
+            previousErrorString === codeEvalResult.getErrorString()) {
+          consecutiveSameRuntimeErrorCounter++;
+        }
+
         var feedback = _getFeedbackWithoutReinforcement(
           tasks, codeEvalResult, rawCodeLineIndexes);
         if (tasks.length > 0 && !codeEvalResult.getErrorString()) {
           feedback.setReinforcement(
             ReinforcementGeneratorService.getReinforcement(
               tasks[tasks.length - 1], codeEvalResult));
         }
+
+        // If the same error counter reaches the threshold, prompt the user to
+        // look at the primer.
+        if (consecutiveSameRuntimeErrorCounter === UNFAMILIARITY_THRESHOLD) {
+          // TODO(dianakc, eledavi): Will have to adjust according to language
+          feedback.appendTextParagraph(_getUnfamiliarLanguageFeedback(
+            LANGUAGE_PYTHON));
+          // Once the user has been prompted, we reset the counter so
+          // that we make sure not to continue to prompt and, thereby,
+          // annoy them.
+          consecutiveSameRuntimeErrorCounter = 0;
+        }
+        previousErrorString = codeEvalResult.getErrorString();
         return feedback;
       },
+      /**
+       * Returns the Feedback object for the given syntax error string.
+       *
+       * @param {string} errorString
+       * @returns {Feedback}
+       */
       getSyntaxErrorFeedback: function(errorString) {
+        // Reset all counters but consecutiveLanguageUnfamiliarityCounter.
+        _resetCounters(EXCLUDE_LANGUAGE_UNFAMILARITY_ERROR);
+
+        // If the user receives another syntax error, increment the
+        // language unfamiliarity error counter.
+        consecutiveLanguageUnfamiliarityCounter++;
         var feedback = FeedbackObjectFactory.create(false);
         feedback.appendSyntaxErrorParagraph(errorString);
+
+        // If the syntax error reaches the threshold, prompt the user to
```

Comment:
I created the _updateCounters() and _checkThreshold() methods based off of your comments. PTAL and LMK your thoughts.